source/speech.py

```diff
@@ -626,7 +640,21 @@ def speakTypedCharacters(ch):
 			log.io("typed word: %s"%typedWord)
 		if config.conf["keyboard"]["speakTypedWords"] and not typingIsProtected:
 			speakText(typedWord)
-	if config.conf["keyboard"]["speakTypedCharacters"] and ord(ch)>=32:
+	global _suppressSpeakTypedCharactersData
+	if _suppressSpeakTypedCharactersData:
+		number, supTime = _suppressSpeakTypedCharactersData
+		# We primarily suppress based on character count,
+		# but time out after a short while just in case.
+		timeOk = time.time() - supTime <= 0.1
+		suppress = number > 0 and timeOk
+		number -= 1
+		if number > 0 and timeOk:

```

Comment:
I think this logic could be cleaned up so the condition does not need to be repeated. It would have the implication that `_suppressSpeakTypedCharactersData` exists for longer, and number is allowed to reach zero.

eg:
```
if _suppressSpeakTypedCharactersData:
	...
	timeOk = time.time() - supTime <= 0.1

	suppress = number > 0 and timeOk
	if suppress:
		_suppressSpeakTypedCharactersData = (--number, supTime)
	else:
		_suppressSpeakTypedCharactersData = None
else:
	suppress = False
if not suppress and config.conf["keyboard"]["speakTypedCharacters"] and ord(ch)>=32:
	speakSpelling(realChar)
```